solv/tdnfpackage.c

```diff
@@ -820,6 +820,58 @@ SolvGetPackageId(
     goto cleanup;
 }
 
+uint32_t
+SolvGetPkgChecksumFromId(
+    PSolvSack pSack,
+    uint32_t dwPkgId,
+    int checksumtype,
+    unsigned char** ppszChecksum)
+{
+
+    uint32_t dwError = 0;
+    const unsigned char* pszTemp = NULL;
+    unsigned char* pszChecksum = NULL;
+    Solvable *pSolv = NULL;
+
+    if(!pSack || !pSack->pPool || !ppszChecksum)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    pSolv = pool_id2solvable(pSack->pPool, dwPkgId);
+    if(!pSolv)
+    {
+        dwError = ERROR_TDNF_NO_DATA;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    pszTemp = solvable_lookup_bin_checksum(pSolv, SOLVABLE_CHECKSUM, &checksumtype);
+
+    if(!pszTemp)
+    {
+        dwError = ERROR_TDNF_NO_DATA;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    dwError = TDNFAllocateString((char *)pszTemp, (char **)&pszChecksum);
```

Comment:
Ah, this is wrong: `solvable_lookup_bin_checksum()` is returning binary data, not a string. You need to figure out the length of the data and use `TDNDAllocateMemory()`, then copy the data.